pkg/reconciler/taskrun/validation_test.go

```diff
@@ -0,0 +1,143 @@
+package taskrun
+
+import (
+	"fmt"
+	"strings"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	pipelinev1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1"
+)
+
+// A helper to create a TaskRun with a platform parameter.
+// It accepts a pointer to a string so we can pass 'nil' to simulate a missing parameter.
+func createTrWithPlatform(platform *string) *pipelinev1.TaskRun {
```

Comment:
We only use this behavior in one specific test, and I'm not a fan of pointers to types to make behavior conditional like this.

Consider special-casing that test out from its test fixture, and don't call this function in its body to make the platform parameter conditional.  We might also change the interface of this function to take a taskrun and return its modified version (or take a pointer and modify it without a copy).